### DISCUSSION TITLE
Alert for simultaneous edits in Admin UI

### DISCUSSION THREAD

**Discussion by runemadsen** [keystonejs/keystone#5313] [Feature requests]
It would be great to have some checking of simultaneous edits in the Admin UI app. So, if a user is editing a list that has been updated since loading the edit page, the save call should be rejected with "This item has been edited by another user since you loaded the page". 

The simplest way to achieve this would probably be to do a MD5 checksum of the GraphQL response, and then have the admin send that along with the mutation, having the GraphQL API make sure that the MD5 checksum matches the current data before saving. That, or Keystone could start enforcing UUID's for every list item that could be used for the same purpose.

Is this something the team would be open for?

**Comment by bladey** [comment]
Hey @runemadsen,

This is a great suggestion and is most likely something we'll implement with our built-in solution for tracking changes and version history when we implement this feature.

Thanks for sharing the idea, we'll keep it in mind!
